lib/board.rb

```diff
@@ -0,0 +1,97 @@
+class Board
+
+ attr_accessor :cells, :moves
+ attr_reader :rank
+
+ WIN_COMBINATIONS = [
+   [0,1,2],
+   [3,4,5],
+   [6,7,8],
+   [0,3,6],
+   [1,4,7],
+   [2,5,8],
+   [0,4,8],
+   [6,4,2]
+ ]
+
+  def initialize(cells = Array.new(9, " "))
+    @cells = cells
+    @moves = []
+  end
+
+  def update(input, player)
+    @cells[input.to_i - 1] = player.token if valid_move?(input)
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    puts "\n=============\n\n"
+  end
+
+  def position(input)
+    @cells[input.to_i - 1]
+  end
+
+  def full?
+    !@cells.include?(" ")
+  end
+
+  def turn_count
+    @cells.select {|c| c != " "}.size
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```